indexer/editable_map_object.cpp

```diff
@@ -228,6 +327,62 @@ bool EditableMapObject::CanUseAsDefaultName(int8_t const lang, vector<int8_t> co
   return false;
 }
 
+void EditableMapObject::RemoveNeedlessNames()
+{
+  if (!IsNamesAdvancedModeEnabled())
+    RemoveFakeNames(m_fakeNames, m_name);
+
+  RemoveBlankNames();
+}
+
+// static
+void EditableMapObject::RemoveFakeNames(FakeNames const & fakeNames, StringUtf8Multilang & name)
+{
+  if (fakeNames.names.empty())
+    return;
+
+  int8_t newDefaultNameCode = StringUtf8Multilang::kUnsupportedLanguageCode;
+  size_t changedCount = 0;
+  string defaultName;
+  name.GetString(StringUtf8Multilang::kDefaultCode, defaultName);
+
+  // New default name calculation priority: 1. name on mwm language, 2. english name.
+  for (int i = fakeNames.names.size() - 1; i >= 0; --i)
```

Comment:
The order of names in fakeNames can be any - no guarantees, right? So it doesn't matter in what order to iterate over names values. If there can be only two values in names (name on mwm language and english name) why not to create two fields instead of vector?